File: LeetCode_Record/2315.统计星号.cpp
#include <string>
using namespace std;

class Solution {
public:
    int countAsterisks(string s) {
        bool start = true;
        int ans = 0;
        for(auto&i : s){
            if(i == '|')
                start = !start;
            else if(start && i == '*')
                ans++;
        }
        return ans;
    }
};

File: LeetCode_Record/2315.统计星号.java
class Solution {
    public int countAsterisks(String s) {
        int n = s.length();
        int ans = 0;
        boolean open = false;
        for(int i = 0; i < n; i++){
            if(s.charAt(i) == '|')
                open = !open;
            else if(open && s.charAt(i) == '*')
                ans++;
        }
        return ans;
    }
}